lib/bolt/config.rb

```diff
@@ -302,6 +302,10 @@ def apply_overrides(options)
         send("#{key}=", options[key]) if options.key?(key)
       end
 
+      if options.key?(:puppetfile)
+        @puppetfile = Pathname.new(File.expand_path(options[:puppetfile], @boltdir.path))
```

Comment:
This should be expanded in the option parser relative to cwd (exactly what we do for `inventoryfile` and `modulepath`). It might be good to standardize those other settings to also return `Pathname` objects so we don't have different behavior depending on whether they came from a Boltdir or not.

Reply:
Updated to expand from cwd and have the parser return a `Pathname` object for `inventoryfile`. It looks like PAL expects a string for `modulepath`, so left that one as-is.